web/src/components/auth/SignUp.tsx

```diff
@@ -0,0 +1,185 @@
+import React, { useState } from "react";
+import TextField from "@material-ui/core/TextField";
+import { styled } from "@material-ui/core/styles";
+import { Auth } from "aws-amplify";
+import Button from "@material-ui/core/Button";
+import CircularProgress from "@material-ui/core/CircularProgress";
+import { Link, useHistory } from "react-router-dom";
+import { Grid } from "@material-ui/core";
+import RightOnLogo from "./RightOnLogo.png";
+
+const DLink = styled(Link)({
+  margin: "15px 0",
+  textAlign: "right",
+});
+
+const Signup: React.FC = () => {
+  const [loading, setLoading] = React.useState(false);
+
+  const history = useHistory();
+
+  const [name, setName] = useState("");
+  const [email, setEmail] = useState("");
+  const [password, setPassword] = useState("");
+  const [confirmPassword, setConfirmPassword] = useState("");
+
+  const handleSignUp = async (e: React.SyntheticEvent<Element, Event>) => {
+    e.preventDefault();
+    setLoading(true);
+
+    if (password !== confirmPassword) {
+      alert("Passwords do not match.");
+      return;
+    }
+    try {
+      await Auth.signUp({
+        username: email,
+        password: confirmPassword,
+        attributes: {
+          email,
+          name,
+        },
+      });
+      alert("Success");
+      history.push("/confirmation");
+    } catch (error) {
+      console.error(error);
+      alert("Error");
+    }
+    setLoading(false);
+  };
+
+  return (
+    <Grid
+      container
+      direction="column"
+      alignItems="center"
+      justifyContent="center"
+    >
+      <img
+        src={RightOnLogo}
```

Comment:
nit: Not important, but we should probably use a vector based image for the logo, e.i. svg, so it supports different scales doesn't get pixelated